contracts/DefifaDeployer.sol

```diff
@@ -198,17 +198,18 @@ contract DefifaDeployer is IDefifaDeployer, IDefifaGamePhaseReporter, IDefifaGam
     /// @return The game phase.
     function currentGamePhaseOf(uint256 _gameId) public view override returns (DefifaGamePhase) {
         // Get the game's current funding cycle along with its metadata.
-        (JBFundingCycle memory _currentFundingCycle, JBFundingCycleMetadata memory _metadata) =
-            controller.currentFundingCycleOf(_gameId);
-
-        if (_currentFundingCycle.number == 0) return DefifaGamePhase.COUNTDOWN;
-        if (_currentFundingCycle.number == 1) return DefifaGamePhase.MINT;
-        if (_noContestIsSet[_gameId]) return DefifaGamePhase.NO_CONTEST;
-        if (_noContestInevitable(_gameId, _currentFundingCycle)) return DefifaGamePhase.NO_CONTEST_INEVITABLE;
-        if (_currentFundingCycle.number == 2 && _opsOf[_gameId].refundPeriodDuration != 0) {
+        (JBRuleset memory _currentRuleset, JBRulesetMetadata memory _metadata) =
+            controller.currentRulesetOf(_gameId);
+
+        if (_currentRuleset.cycleNumber == 0) return DefifaGamePhase.COUNTDOWN;
+        if (_currentRuleset.cycleNumber == 1) return DefifaGamePhase.MINT;
+        // TODO: Do we need these two states?
```

Comment:
we did for some reason, i remember. but lets default remove and see if we need to add them back. just starting the review here.